File: backend/models/payment.py
from utils.db import db
from datetime import datetime

class Payment(db.Model):
    __tablename__ = 'payments'
    id = db.Column(db.Integer, primary_key=True)
    user_id = db.Column(db.Integer, db.ForeignKey('users.id', ondelete="CASCADE"), nullable=False)
    hospital_id = db.Column(db.Integer, db.ForeignKey('hospitals.id'), nullable=True)
    amount = db.Column(db.Float, nullable=False)
    transaction_id = db.Column(db.String(100), unique=True, nullable=False)
    purpose = db.Column(db.String(100), nullable=False) # e.g., 'Hospital Registration Fee'
    status = db.Column(db.String(20), default='Successful')
    created_at = db.Column(db.DateTime, default=datetime.utcnow)

    user = db.relationship('User', backref=db.backref('payments', cascade="all, delete-orphan"))
    hospital = db.relationship('Hospital', backref='payments')

    def to_dict(self):
        return {
            'id': self.id,
            'user_id': self.user_id,
            'hospital_id': self.hospital_id,
            'amount': self.amount,
            'transaction_id': self.transaction_id,
            'purpose': self.purpose,
            'status': self.status,
            'created_at': self.created_at.isoformat()
        }

File: backend-node/models/Payment.js
const mongoose = require('mongoose');

const paymentSchema = new mongoose.Schema({
  user_id: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
  hospital_id: { type: mongoose.Schema.Types.ObjectId, ref: 'Hospital', default: null },
  amount: { type: Number, required: true },
  transaction_id: { type: String, required: true, unique: true },
  purpose: { type: String, required: true },
  status: { type: String, default: 'Successful' },
  created_at: { type: Date, default: Date.now }
});

paymentSchema.set('toJSON', {
  transform: (document, returnedObject) => {
    returnedObject.id = returnedObject._id.toString();
    returnedObject.user_id = returnedObject.user_id ? returnedObject.user_id.toString() : null;
    returnedObject.hospital_id = returnedObject.hospital_id ? returnedObject.hospital_id.toString() : null;
    delete returnedObject._id;
    delete returnedObject.__v;
  }
});

module.exports = mongoose.model('Payment', paymentSchema);
